sw/device/silicon_creator/manuf/lib/individualize_sw_cfg.c

```diff
@@ -91,6 +91,44 @@ static status_t otp_img_write(const dif_otp_ctrl_t *otp,
   return OK_STATUS();
 }
 
+/**
+ * Reads out the expected values of the fields that are not provisioned until
+ * the end of personalization process.
+ *
+ * @param partition Target OTP partition.
+ * @param kv OTP Array of OTP key values. See `otp_kv_t` documentation for more
+ * details.
+ * @param len Length of the `kv` array.
+ * @param buffer A buffer containing the entire target OTP partition.
+ * @return OK_STATUS if the expected OTP values are successfully written to the
+ * `buffer`.
+ */
+static status_t otp_img_expected_value_read(dif_otp_ctrl_partition_t partition,
+                                            const otp_kv_t *kv, size_t len,
+                                            uint8_t *buffer) {
+  for (size_t i = 0; i < len; ++i) {
+    if (kv[i].offset != OTP_CTRL_PARAM_OWNER_SW_CFG_ROM_BOOTSTRAP_DIS_OFFSET) {
```

Comment:
Maybe this offset could be a passed parameter? so this would more easily generalize in the future?

Reply:
Also, I feel like this function is necessarily complicated. Cant the function just take the following inputs:
- the buffer
- the kv item that should be overwritten
and just overwrite that field? why do you need to loop over the whole buffer? Am I missing something?

Reply:
> why do you need to loop over the whole buffer? Am I missing something?

We don't know where a given field is located in the kv array, so we need to loop over the kv array to find the corresponding kv item by checking the offset.
> Maybe this offset could be a passed parameter? so this would more easily generalize in the future?

I did consider passing the offset as a parameter, but since we have to loop over the kv array to find a target kv item, sending the offsets under the same partition separately will require looping over the kv array multiple times to update the buffer, while maintaining the offsets in this function only requires looping the kv array at most once  to update the buffer of a given partition.

Reply:
Ah, I just realized that since we've already exposed the expected value of the fields that are not provisioned until the end of perso, it's actually unnecessary to get the expected value by accessing the kv item.

Reply:
Done.